### PR TITLE
Fix searches from handbooks.

### LINKED ISSUE #183: Fix searches from handbooks.
If you search using the search control on the handbook template, you are taken to functions search results. It should search relative to the handbooks. 

For example :`/?s=keyword` -> `/blocks-editor/?s=keyword`.

As it sits now, if you visit  `/blocks-editor/?s=keyword`, the search results template will pick it up and it will look like:

Screen Shot 2023-01-30 at 5 34 51 PM

There isn't currently a design for those results although we can probably just remove the grey tab and then request feedback from design.

### PR DESCRIPTION
**Pull request by StevenDufresne**
Fixes #183

This PR fixes searches for handbooks.

## Code Search
localhost_8888__s=get 

## Rest API Search
localhost_8888_rest-api__s=get 

## Block Editor 
localhost_8888_block-editor__s=get

### CODE REVIEW

**Comment by ryelle** [comment]
Tested a few handbook searches & the code ref search. Working well, but I noticed a few little things.

It looks like the styles in `search-title/style.scss` & `search-results-context/style.scss` are exactly the same, so it's functionally loading duplicate CSS.

On paginated pages, the "Result" item in the breadcrumbs links to the main reference search page (for example, on http://localhost:8888/block-editor/page/2/?s=block.json it links to http://localhost:8888/search/block.json/).

Screenshot 2023-02-03 at 1 34 10 PM

Would it be possible to style the article title with the serif font when it's not a code reference response? And maybe remove the `break-all` style.

Screenshot 2023-02-03 at 1 48 25 PM

I managed to trigger the autocomplete on a handbook search result page:

Screenshot 2023-02-03 at 1 47 00 PM

**Comment by StevenDufresne** [comment]
> It looks like the styles in `search-title/style.scss` & `search-results-context/style.scss` are exactly the same, so it's functionally loading duplicate CSS.

Eeek. Removed.

> On paginated pages, the "Result" item in the breadcrumbs links to the main reference search page (for example, on http://localhost:8888/block-editor/page/2/?s=block.json it links to http://localhost:8888/search/block.json/).

Nice find. For the record, it appears like the non-handbook search link has a rewrite `search/{term}`. Handbooks don't.

> Would it be possible to style the article title with the serif font when it's not a code reference response? And maybe remove the `break-all` style.

Done.

> I managed to trigger the autocomplete on a handbook search result page:
> 

Okay, decided to no enqueue if not handbook. [46112a4](https://github.com/WordPress/wporg-developer/pull/195/commits/46112a4f9aace3c4a977b47dc93f3e9cd11ae4ef)

File: source/wp-content/themes/wporg-developer-2023/inc/breadcrumb-trail.php [hunk 1]
```diff
@@ -833,6 +837,24 @@ public function do_search_items() {
...
+	 * @since  0.6.0
+	 * @access public
+	 * @return void
+	 */
+	public function do_handbook_search_items() {
+		$this->items[] = '<a href="' . wporg_get_current_handbook_home_url() . '">' . get_queried_object()->label . '</a>';
+
+		if ( is_paged() ) {
+			$search_url = sprintf( '%s?s=%s', wporg_get_current_handbook_home_url(), get_search_query() );
+			$this->items[] = '<a href="' . esc_attr( $search_url ) . '">' . esc_html( $this->args['labels']['search'] ) . '</a>';
```
Comment (ryelle) [inline source/wp-content/themes/wporg-developer-2023/inc/breadcrumb-trail.php]:
```suggestion
			$this->items[] = '<a href="' . esc_url( $search_url ) . '">' . esc_html( $this->args['labels']['search'] ) . '</a>';
```

File: source/wp-content/themes/wporg-developer-2023/inc/breadcrumb-trail.php [hunk 2]
```diff
@@ -833,6 +837,24 @@ public function do_search_items() {
...
+	 *
+	 * @since  0.6.0
+	 * @access public
+	 * @return void
+	 */
+	public function do_handbook_search_items() {
+		$this->items[] = '<a href="' . wporg_get_current_handbook_home_url() . '">' . get_queried_object()->label . '</a>';
+
+		if ( is_paged() ) {
+			$search_url = sprintf( '%s?s=%s', wporg_get_current_handbook_home_url(), get_search_query() );
```
Comment (ryelle) [inline source/wp-content/themes/wporg-developer-2023/inc/breadcrumb-trail.php]:
```suggestion
			$search_url    = add_query_arg( 's', get_search_query(), wporg_get_current_handbook_home_url() );
```

I think it's a better practice to use `add_query_arg`, even if you're pretty sure that `wporg_get_current_handbook_home_url` will return a plain URL.

**Review by ryelle — APPROVED** [review]
Two pretty minor suggestions, once those are addressed this is good to merge 👍🏻

**Pull request merged**